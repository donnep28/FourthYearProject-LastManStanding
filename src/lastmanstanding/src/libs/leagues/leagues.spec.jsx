import React from 'react';
import {shallow} from 'enzyme';
import Leagues from './leagues';
import renderer from "react-test-renderer";
import axios from 'axios';

jest.mock('axios');

const mockSetState = jest.fn();

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useState: (initial) => [initial, mockSetState],
}));

describe('leagues', () => {
  beforeAll(() => {
    jest.resetAllMocks();
  });

  const props = {
    user: {
      attributes: {
        sub: 'ecbc56df-d81a-410d-b399-4421b55ced32',
      },
    },
    results: {},
  };
  describe('Snapshot', () => {
    it('renders undefined user', () => {
      const props = {
        user: {},
        results: {},
      };
      const component = shallow(
          <Leagues {...props}/>,
      );
      expect(component).toMatchSnapshot();
    });
  });
  it('renders defined user', async () => {
    const myLeaguesInfo = [
      [
        {
          Admin: 'Yes',
          CurrentPick: 'Everton FC',
          LeagueID: 'Plips League#283625',
          // eslint-disable-next-line max-len
          LeaguePlayerID: 'Plips League#283625/ecbc56df-d81a-410d-b399-4421b55ced32',
          PickedTeams: ['Everton FC'],
          // eslint-disable-next-line max-len
          UnpickedTeams: ['Manchester United FC', 'Manchester City FC', 'Leicester City FC'],
          Username: 'philip',
          createdTime: '2021-02-22 16:46:34.558082',
          fullName: 'philip donnelly',
          playerStatus: 'In',
        },
      ],
    ];
    axios.get.mockImplementationOnce(() => Promise.resolve(
        {data: myLeaguesInfo},
    ));
    const tree = renderer.create(<Leagues {...props}/>);
    await Promise.resolve();
    expect(tree.toJSON()).toMatchSnapshot();
  });

  describe('Button Clicks', () => {
    it('updates join league button', () => {
      const component = shallow(
          <Leagues {...props}/>,
      );
      component.find('[data-automation="JoinLeague"]').simulate('click');
      expect(mockSetState).toHaveBeenCalled();
    });
    it('updates create league button', () => {
      const component = shallow(
          <Leagues {...props}/>,
      );
      component.find('[data-automation="CreateLeague"]').simulate('click');
      expect(mockSetState).toHaveBeenCalled();
    });
  });
});
