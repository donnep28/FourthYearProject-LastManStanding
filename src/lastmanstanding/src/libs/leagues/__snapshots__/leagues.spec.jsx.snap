// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`leagues Snapshot renders undefined user 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Grid))
    container={true}
    direction="row"
    spacing={2}
  >
    <WithStyles(ForwardRef(Grid))
      align="center"
      item={true}
      md={8}
      xs={12}
    >
      <WithStyles(ForwardRef(Grid))
        className="buttons"
        item={true}
        md={8}
        xs={12}
      >
        <WithStyles(ForwardRef(Button))
          data-automation="CreateLeague"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "white",
              "border": "4px solid #490050",
              "color": "#490050",
              "marginBottom": "5px",
              "marginTop": "10px",
              "width": 250,
            }
          }
          variant="contained"
        >
          <Memo(ForwardRef(AddCircleOutlineIcon)) />
          <h4
            style={
              Object {
                "paddingLeft": "5px",
              }
            }
          >
            Create League
          </h4>
        </WithStyles(ForwardRef(Button))>
        <WithStyles(ForwardRef(Button))
          data-automation="JoinLeague"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "white",
              "border": "4px solid #490050",
              "color": "#490050",
              "marginBottom": "5px",
              "marginTop": "10px",
              "width": 250,
            }
          }
          variant="contained"
        >
          <Memo(ForwardRef(PeopleAltIcon)) />
          <h4
            style={
              Object {
                "paddingLeft": "5px",
              }
            }
          >
            Join League
          </h4>
        </WithStyles(ForwardRef(Button))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        md={8}
        xs={12}
      >
        <div>
          <div
            className="leaguesLoading"
          >
            <WithStyles(ForwardRef(CircularProgress))
              style={
                Object {
                  "color": "#490050",
                }
              }
            />
          </div>
        </div>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      md={4}
      xs={12}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        style={
          Object {
            "marginBottom": "20px",
          }
        }
        xs={12}
      >
        <PremierLeagueFixtures
          results={Object {}}
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <PremierLeagueResults
          results={Object {}}
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</Fragment>
`;

exports[`leagues renders defined user 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
>
  <div
    align="center"
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8"
  >
    <div
      className="MuiGrid-root buttons MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained"
        data-automation="CreateLeague"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "border": "4px solid #490050",
            "color": "#490050",
            "marginBottom": "5px",
            "marginTop": "10px",
            "width": 250,
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
            />
          </svg>
          <h4
            style={
              Object {
                "paddingLeft": "5px",
              }
            }
          >
            Create League
          </h4>
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained"
        data-automation="JoinLeague"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "border": "4px solid #490050",
            "color": "#490050",
            "marginBottom": "5px",
            "marginTop": "10px",
            "width": 250,
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.67 13.13C18.04 14.06 19 15.32 19 17v3h4v-3c0-2.18-3.57-3.47-6.33-3.87z"
              fillRule="evenodd"
            />
            <circle
              cx="9"
              cy="8"
              fillRule="evenodd"
              r="4"
            />
            <path
              d="M15 12c2.21 0 4-1.79 4-4s-1.79-4-4-4c-.47 0-.91.1-1.33.24C14.5 5.27 15 6.58 15 8s-.5 2.73-1.33 3.76c.42.14.86.24 1.33.24zM9 13c-2.67 0-8 1.34-8 4v3h16v-3c0-2.66-5.33-4-8-4z"
              fillRule="evenodd"
            />
          </svg>
          <h4
            style={
              Object {
                "paddingLeft": "5px",
              }
            }
          >
            Join League
          </h4>
        </span>
      </button>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8"
    >
      <div>
        <div
          className="leaguesLoading"
        >
          <div
            className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
            role="progressbar"
            style={
              Object {
                "color": "#490050",
                "height": 40,
                "width": 40,
              }
            }
          >
            <svg
              className="MuiCircularProgress-svg"
              viewBox="22 22 44 44"
            >
              <circle
                className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                cx={44}
                cy={44}
                fill="none"
                r={20.2}
                strokeWidth={3.6}
                style={Object {}}
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      style={
        Object {
          "marginBottom": "20px",
        }
      }
    >
      <div />
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div />
    </div>
  </div>
</div>
`;
